modules/azure/compute.go

```diff
@@ -8,97 +8,451 @@ import (
 }
 
-// GetTagsForVirtualMachine gets the tags of the given Virtual Machine as a map.
+// GetVirtualMachineTags gets the Tags of the specified Virtual Machine as a map.
 // This function would fail the test if there is an error.
-func GetTagsForVirtualMachine(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) map[string]string {
-	tags, err := GetTagsForVirtualMachineE(t, vmName, resGroupName, subscriptionID)
+func GetVirtualMachineTags(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) map[string]string {
+	tags, err := GetVirtualMachineTagsE(vmName, resGroupName, subscriptionID)
 	require.NoError(t, err)
 
 	return tags
 }
 
-// GetTagsForVirtualMachineE gets the tags of the given Virtual Machine as a map
-func GetTagsForVirtualMachineE(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) (map[string]string, error) {
+// GetVirtualMachineTagsE gets the Tags of the specified Virtual Machine as a map.
+func GetVirtualMachineTagsE(vmName string, resGroupName string, subscriptionID string) (map[string]string, error) {
 	// Setup a blank map to populate and return
-	tags := make(map[string]string)
+	var tags map[string]string
+
+	// Get VM Object
+	vm, err := GetVirtualMachineE(vmName, resGroupName, subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Range through existing tags and populate above map accordingly
+	for k, v := range vm.Tags {
+		tags[k] = *v
+	}
+
+	return tags, nil
+}
+
+// ***************************************************** //
+// Get multiple Virtual Machines from a Resource Group
+// ***************************************************** //
+
+// ListVirtualMachinesForResourceGroup gets a list of all Virtual Machine names in the specified Resource Group.
+// This function would fail the test if there is an error.
+func ListVirtualMachinesForResourceGroup(t testing.TestingT, resGroupName string, subscriptionID string) []string {
+	vms, err := ListVirtualMachinesForResourceGroupE(resGroupName, subscriptionID)
+	require.NoError(t, err)
+	return vms
+}
+
+// ListVirtualMachinesForResourceGroupE gets a list of all Virtual Machine names in the specified Resource Group.
+func ListVirtualMachinesForResourceGroupE(resourceGroupName string, subscriptionID string) ([]string, error) {
+	var vmDetails []string
+
+	vmClient, err := GetVirtualMachineClientE(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	vms, err := vmClient.List(context.Background(), resourceGroupName)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, v := range vms.Values() {
+		vmDetails = append(vmDetails, *v.Name)
+	}
+	return vmDetails, nil
+}
+
+// GetVirtualMachinesForResourceGroup gets all Virtual Machine objects in the specified Resource Group.
+// This function would fail the test if there is an error.
+func GetVirtualMachinesForResourceGroup(t testing.TestingT, resGroupName string, subscriptionID string) *map[string]compute.VirtualMachineProperties {
```

Comment:
Don't use pointers with map in golang, because it's already a pointer: https://blog.golang.org/maps

> Map types are reference types, like pointers or slices

Reply:
Updated, thanks for the link, helpful to know!